source/core/slang-relative-container.h

```diff
@@ -0,0 +1,254 @@
+// slang-relative-container.h
+#ifndef SLANG_RELATIVE_CONTAINER_H_INCLUDED
+#define SLANG_RELATIVE_CONTAINER_H_INCLUDED
+
+#include "slang-basic.h"
+
+namespace Slang {
+
+struct RelativeBase
+{
+    uint8_t* m_data;
+
+    static RelativeBase g_null;
+};
+
+template <typename T>
+class Safe32Ptr
```

Comment:
I'm maybe missing the point of this type right now. The relative pointer type is clear to me, but I'm not seeing what is going to be `Safe` about this one, since there doesn't seem to be any bounds-checking or anything.

Reply:
Yes - it's for construction where the underlying containers memory is moved typically to make more space available.